test/pylib/scylla_cluster.py

```diff
@@ -698,6 +712,25 @@ def update_config(self, server_id: ServerNum, key: str, value: object) -> Action
         self.servers[server_id].update_config(key, value)
         return ScyllaCluster.ActionReturn(success=True)
 
+    async def change_ip(self, server_id: ServerNum) -> ActionReturn:
+        """Lease a new IP address and update conf/scylla.yaml with it. The
+        original IP is released at the end of the test to avoid recycle
+        within the existing cluster (scylla currently doesn't support
```

Comment:
I think we're talking about different scenarios here. One scenario is when we, say, have a cluster of 3 instances. We stop the instances and then start them having swapped or otherwise rotated their IP addresses. I think that write coordinators and drivers will go crazy when they see that, since they typically associate tokens with ip address, not host id. To correctly test that (with a failing test) we will need to build a correctness test, including an LWT test. I also don't see this being the goal right now. 

Please note that with skip_wait_for_gossip_to_settle set to 0 the above scenario doesn't have to be coordinated - we can simply start a bunch of new nodes and change IPs of a bunch of other nodes - and naturally some address will be recycled before tokens are reassigned and things will be screwed.

Another matter is when an IP address that was used is later recycled for a different node, after a few gossip rounds and information about the previous IP address change has settled. This can work - but then again I would have to sleep in the test and wait until the node is UP and all drivers have been updated. Once again, please keep in mind drivers are tracking the nodes using IP addresses so they may miss a gossip notification and route a write to a wrong node, assuming it owns the ranges of the node that has got its IP address changed. 

I don't see a point devising a failing test - because things can go wrong in many different ways. I see a point in switching to coordinator-based topology changes - first, then - to changing the drivers across the board to become aware of topology fencing tokens - so that they never use the wrong topology in the first place - and then covering up this works correctly with a randomized test. 

Spending time on a failing test until all bits and pieces have fallen into their place is falling into a rabbit hole.

Reply:
I agree it's out of scope of this PR to test all that, but the comment should be accurate nevertheless.

Reply:
I removed the claim that "Scylla doesn't support this".